release/src/main/groovy/TestScripts.groovy

```diff
@@ -78,57 +116,57 @@ class TestScripts {
      shell[2] = cmd
      def pb = new ProcessBuilder(shell)
      pb.directory(var.curDir)
+     pb.redirectErrorStream(true)
      def proc = pb.start()
      var.lastText = ""
+     def text = StringBuilder.newInstance()
      proc.inputStream.eachLine {
        println it
-       var.lastText += it + "\n";
+       text.append(it + "\n")
      }
-     var.lastText = var.lastText.trim()
      proc.waitFor()
+     var.lastText = text.toString().trim()
      if (proc.exitValue() != 0) {
        println var.lastText
        _error("Failed command")
      }
    }
-   
+
    // Change directory
    private void _chdir(String subdir) {
      var.curDir = new File(var.curDir.absolutePath, subdir)
      if (!var.curDir.exists()) {
        _error("Directory ${var.curDir} not found")
      }
-     _execute("pwd")
-     if (var.lastText != var.curDir.absolutePath) {
-       _error("Directory mismatch, ${var.lastText} != ${var.curDir.absolutePath}")
-   
-     }
    }
-   
-   // Run a maven command, setting up a new local repository and a settings.xml with the snapshot repository
+
+   // Run a maven command, setting up a new local repository and a settings.xml with a custom repository
    private void _mvn(String args) {
      def m2 = new File(var.startDir, ".m2/repository")
      m2.mkdirs()
      def settings = new File(var.startDir, "settings.xml")
-     def repo = System.env.snapshot_repository ?: "https://repository.apache.org/content/repositories/snapshots"
      settings.write """
        <settings>
          <localRepository>${m2.absolutePath}</localRepository>
            <profiles>
              <profile>
-               <id>snapshot</id>
+               <id>testrel</id>
                  <repositories>
                    <repository>
-                     <id>apache.snapshots</id>
-                     <url>${repo}</url>
+                     <id>test.release</id>
+                     <url>${var.repoUrl}</url>
                    </repository>
                  </repositories>
                </profile>
              </profiles>
         </settings>
         """
-       def cmd = "mvn ${args} -s${settings.absolutePath} -Psnapshot -B"
-       _execute(cmd)
+       def cmd = "mvn ${args} -s ${settings.absolutePath} -Ptestrel -B"
+       String path = System.getenv("PATH");
+       // Set the path on jenkins executors to use 3.5.2.
+       def mvnPath = "/home/jenkins/tools/maven/apache-maven-3.5.2/bin"
```

Comment:
Use ${MAVEN_HOME}/bin as per https://cwiki.apache.org/confluence/display/INFRA/Maven+Installation+Matrix

Reply:
Done

Reply:
Just ran it on beam2, and MAVEN_HOME was not set?